universal-application-tool-0.0.1/app/views/admin/programs/ProgramBlockPredicatesEditView.java

```diff
@@ -105,6 +111,39 @@ private Modal predicateFormModal(
 
   private Tag renderPredicateForm(
       String blockName, QuestionDefinition questionDefinition, Tag csrfTag) {
+
+    ImmutableMap<Scalar, ScalarType> scalars;
+    try {
+      scalars = Scalar.getScalars(questionDefinition.getQuestionType());
+    } catch (InvalidQuestionTypeException | UnsupportedQuestionTypeException e) {
+      return div()
+          .withText("Sorry, you cannot create a show/hide predicate with this question type.");
+    }
+
+    ImmutableList.Builder<SimpleEntry<String, String>> scalarOptionsBuilder =
+        ImmutableList.builder();
+    scalars.forEach(
+        (scalar, type) -> {
+          scalarOptionsBuilder.add(new SimpleEntry<>(scalar.toString(), scalar.toString()));
```

Comment:
question (blocking): If the html value is the human-readable string, how do we plan to convert back to the enum on form submission?

Reply:
Yeah, good point. I think this is the wrong way to do this and that I need to use scalar.name() for the "key" in the option, even if we use a display string for the name. That should work right?

Related: I'm gonna update the `Scalar` enum. Instead of having that toString method with the case statements, I'll make it look more like the `Operator` enum by having it take a displayString.